dvc/repo/__init__.py

```diff
@@ -335,19 +337,17 @@ def graph(self, stages=None, from_directory=None):
             for dep in stage.deps:
                 for out in outs:
                     if (
-                        out.path_info != dep.path_info
-                        and not dep.path_info.isin(out.path_info)
-                        and not out.path_info.isin(dep.path_info)
+                        out == dep.path_info
+                        or dep.path_info.isin(out)
+                        or out.isin(dep.path_info)
                     ):
-                        continue
-
-                    dep_stage = out.stage
-                    dep_node = relpath(dep_stage.path, self.root_dir)
-                    G.add_node(dep_node, stage=dep_stage)
-                    G.add_edge(node, dep_node)
-                    if not stage.locked:
-                        G_active.add_node(dep_node, stage=dep_stage)
-                        G_active.add_edge(node, dep_node)
+                        dep_stage = outs.get(out).stage
```

Comment:
```suggestion
                        dep_stage = outs[out].stage
```
This way we will get appropriate `KeyError` instead of `None does not have attribute "stage"`, same as above.

Reply:
👍